flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/AbstractDeclarativeSlotPoolBridgeTest.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.flink.util.clock.SystemClock;
+
+import javax.annotation.Nonnull;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.apache.flink.runtime.concurrent.ComponentMainThreadExecutorServiceAdapter.forMainThread;
+
+/** Test base class for {@link DeclarativeSlotPoolBridge}. */
+abstract class AbstractDeclarativeSlotPoolBridgeTest {
+
+    protected static final Duration rpcTimeout = Duration.ofSeconds(20);
+    protected static final JobID jobId = new JobID();
+    protected static final JobMasterId jobMasterId = JobMasterId.generate();
+    protected static final ComponentMainThreadExecutor mainThreadExecutor = forMainThread();
+
+    @Parameter protected RequestSlotMatchingStrategy requestSlotMatchingStrategy;
+
+    @Parameter(1)
+    protected Duration slotRequestMaxInterval;
+
+    @Parameters(name = "requestSlotMatchingStrategy: {0}, slotRequestMaxInterval: {1}")
+    static Collection<Object[]> data() throws IOException {
+        return Arrays.asList(
+                new Object[] {SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO},
+                new Object[] {SimpleRequestSlotMatchingStrategy.INSTANCE, Duration.ofMillis(50)},
+                new Object[] {
+                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE, Duration.ZERO
+                },
+                new Object[] {
+                    PreferredAllocationRequestSlotMatchingStrategy.INSTANCE, Duration.ofMillis(50)
+                });
+    }
+
+    @Nonnull
+    static DeclarativeSlotPoolBridge createDeclarativeSlotPoolBridge(
+            DeclarativeSlotPoolFactory declarativeSlotPoolFactory,
+            RequestSlotMatchingStrategy requestSlotMatchingStrategy,
+            Duration slotRequestMaxInterval) {
+        return new DeclarativeSlotPoolBridge(
+                jobId,
+                declarativeSlotPoolFactory,
+                SystemClock.getInstance(),
+                rpcTimeout,
+                Duration.ofSeconds(20),
+                Duration.ofSeconds(20),
+                requestSlotMatchingStrategy,
+                slotRequestMaxInterval,
+                mainThreadExecutor);
+    }
+
+    @Nonnull
+    static DeclarativeSlotPoolBridge createDeclarativeSlotPoolBridge(
+            DeclarativeSlotPoolFactory declarativeSlotPoolFactory,
+            RequestSlotMatchingStrategy requestSlotMatchingStrategy,
+            Duration slotRequestMaxInterval,
+            ComponentMainThreadExecutor mainThreadExecutor) {
+        return new DeclarativeSlotPoolBridge(
+                jobId,
+                declarativeSlotPoolFactory,
+                SystemClock.getInstance(),
+                rpcTimeout,
+                Duration.ofSeconds(20),
+                Duration.ofSeconds(20),
+                requestSlotMatchingStrategy,
+                slotRequestMaxInterval,
+                mainThreadExecutor);
+    }
```

Comment:
The first `createDeclarativeSlotPoolBridge` could call the second `createDeclarativeSlotPoolBridge` to new DeclarativeSlotPoolBridge.